cmd/hope/vm/image.go

```diff
@@ -0,0 +1,180 @@
+package vm
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+)
+
+import (
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+import (
+	"github.com/Eagerod/hope/cmd/hope/utils"
+	"github.com/Eagerod/hope/pkg/packer"
+	"github.com/Eagerod/hope/pkg/scp"
+	"github.com/Eagerod/hope/pkg/ssh"
+)
+
+var imageCmdParameterSlice *[]string
+
+func initImageCmdFlags() {
+	imageCmdParameterSlice = imageCmd.Flags().StringArrayP("param", "p", []string{}, "parameters to forward to packer's -var")
+}
+
+type PackerBuilder struct {
+	VMName          string `json:"vm_name"`
+	OutputDirectory string `json:"output_directory"`
+}
+
+type PackerSpec struct {
+	Builders []PackerBuilder
+}
+
+var imageCmd = &cobra.Command{
+	Use:   "image",
+	Short: "Creates a VM image from the defined packer spec.",
+	Args:  cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		vmName := args[0]
+
+		vms, err := utils.GetVMs()
+		if err != nil {
+			return err
+		}
+
+		vm, err := utils.VMSpec(vmName)
+		if err != nil {
+			return err
+		}
+
+		vmDir := path.Join(vms.Root, vm.Name)
+		log.Trace(fmt.Sprintf("Looking for VM definition in %s", vmDir))
+
+		stat, err := os.Stat(vmDir)
+		if err != nil && os.IsNotExist(err) {
```

Comment:
Tighten up these three conditions into a single group, so the `stat` doesn't keep scope?

Reply:
Or get rid of the last validation at the bottom, seems pretty extra, and the packer.json check below will do that work anyways.